src/simplify_reshapes.cpp

```diff
@@ -122,6 +122,11 @@ struct find_nop_reshapes
         reshapes.insert("pad");
         reshapes.insert("slice");
         reshapes.insert("transpose");
+        reshapes.insert("reduce_mean");
```

Comment:
Would be better to assert that axes are either empty resulting in no-op or 
axes  lengths are 1.